crates/ev-cli/src/commands/function/deploy.rs

```diff
@@ -0,0 +1,237 @@
+            DeployError::DeploymentFailed | DeployError::DeploymentCancelled => {
+                "function-deployment-error"
+            }
+            DeployError::DeploymentStatusFetch(_) => "function-deployment-status-fetch-error",
+        }
+        .to_string()
+    }
+
+    fn exitcode(&self) -> crate::errors::ExitCode {
+        crate::errors::GENERAL
+    }
+}
+
+#[derive(strum_macros::Display, Debug)]
+pub enum DeployMessage {
+    #[strum(to_string = "Function ({uuid}) Deployed Succesfully")]
+    Deployed { uuid: String },
+    #[strum(
+        to_string = "Function deployment initiated successfully. Deployment will continue in the background."
+    )]
+    BackgroundDeployment,
+}
+
+impl CmdOutput for DeployMessage {
+    fn code(&self) -> String {
+        match self {
+            DeployMessage::Deployed { .. } => "function-deployed",
+            DeployMessage::BackgroundDeployment => "function-background-deployment-started",
+        }
+        .to_string()
+    }
+
+    fn exitcode(&self) -> crate::errors::ExitCode {
+        crate::errors::OK
+    }
+}
+
+pub async fn run(args: DeployArgs, auth: BasicAuth) -> Result<DeployMessage, DeployError> {
+    let api_client = papi::EvApiClient::new(auth);
+    let base_args = BaseArgs::parse();
+    let is_quiet = !base_args.verbose || base_args.json;
+
+    crate::fs::validate_function_directory_structure()?;
+    crate::fs::validate_function_toml()?;
+
+    let name = get_current_function_name().map_err(|_| DeployError::MissingNameField)?;
+
+    validators::validate_function_name(&name)?;
+
+    let language = get_current_function_language()?;
+
+    validate_function_language(&language)?;
+
+    let current_date: NaiveDate = Utc::now().date_naive();
+
+    if let Some(&deprecation_date) = LANGUAGE_DEPRECATION_DATE_MAP.get(&language) {
+        return Err(if current_date > deprecation_date {
+            DeployError::VersionDeprecated(deprecation_date, language)
+        } else {
+            DeployError::VersionWillBeDeprecated(deprecation_date, language)
+        });
+    }
+
+    let tmp_dir = TempDir::new().unwrap();
+
+    let progress = interact::start_spinner("Zipping current direction...", is_quiet);
+
+    let destination = match zip_current_directory(&name, tmp_dir.path()) {
+        Ok(destination) => destination,
+        Err(e) => {
+            progress.finish_with_message(
+                "An error occurred while zipping your Function source.".into(),
+            );
+            return Err(DeployError::FsError(e));
+        }
+    };
+
+    progress.finish_with_message("Directory zipped succesfully".into());
+
+    let progress = interact::start_spinner("Beginning Function deployment...", is_quiet);
+
+    let apps_functions = api_client
+        .get_all_functions_for_app()
+        .await
+        .map_err(DeployError::FetchAppFunctions)?;
+    let maybe_function = apps_functions.iter().find(|f| f.name == name);
```

Comment:
Did we add an endpoint in the end to get by name?

Reply:
yea I have a PR on a local branch, can push it. Decided it's a waste of time now tho, number of functions is usually relatively low so it's not actually as bad as it looks. Just want to unblock myself from other tasks